tsl/test/t/006_job_crash_log.pl

```diff
@@ -0,0 +1,64 @@
+# This file and its contents are licensed under the Timescale License.
+# Please see the included NOTICE for copyright information and
+# LICENSE-TIMESCALE for a copy of the license.
+
+use strict;
+use warnings;
+use TimescaleNode;
+use Test::More;
+use TestLib;
+
+# This test checks that a job crash is reported in the job errors table
+# (ie a record gets inserted into the table _timescaledb_internal.job_errors).
+# We cannot do that with a regression test because the server will not recover after
+# a crash in that case
+
+my $node = TimescaleNode->create();
+
+# by default PostgresNode doesn't doesn't restart after a crash
+# taken from 013_crash_restart.pl
+$node->safe_psql(
+	'postgres',
+	q[ALTER SYSTEM SET restart_after_crash = 1;
+				   ALTER SYSTEM SET log_connections = 1;
+				   SELECT pg_reload_conf();]);
+
+# create proc to run as job
+my $query = <<'END_OF_SQL';
+CREATE OR REPLACE PROCEDURE custom_proc_sleep60(jobid int, config jsonb) LANGUAGE PLPGSQL AS
+$$
+BEGIN
+ RAISE NOTICE 'im about to sleep for 60 seconds, plenty of time for you to kill me';
+ perform pg_sleep(60);
+END
+$$;
+END_OF_SQL
+
+my $ret = $node->safe_psql('postgres', "$query");
+
+my $query_add =
+  q[select add_job('custom_proc_sleep60', '5 minutes', initial_start => now())];
+my $jobid = $node->safe_psql('postgres', "$query_add");
+is($jobid, '1000', 'job was added');
+
+# sleep 10 to make sure job has started running
+$node->safe_psql('postgres', "select pg_sleep(10)");
+# select the pid of this job in order to kill it
+my $query_pid =
+  qq[select pid from pg_stat_activity where application_name like 'User-Defined Action%' and query like '%custom_proc_sleep60%'];
```

Comment:
Heredocs work here as well. Perl is great in that way. Note the use of double-quotes for the heredoc tag.

```suggestion
my $query_pid = <<"END_OF_QUERY";
select pid from pg_stat_activity
where application_name like 'User-Defined Action%'
   and query like '%custom_proc_sleep60%'
END_OF_QUERY
```

Reply:
I'm not sure you actually need the double-quotes here since you have no Perl variables in the query.